javascript/modules/client/src/client.ts

```diff
@@ -282,32 +277,93 @@ export class Client extends ClientCore implements IClient {
-    ).then((gasLimit) => {
-      return this.web3.getApproximateGasFee(gasLimit.toBigInt());
+    return daoInstance.estimateGas
+      .deposit(tokenAddress, amount, reference, override)
+      .then(gasLimit => {
+        return this.web3.getApproximateGasFee(gasLimit.toBigInt());
+      });
+  }
+
+  //// PRIVATE METHODS METADATA
+
+  private async _getDaoMetadata(daoIdentifier: string): Promise<DaoMetadata> {
+    // TODO: Implement actual fetch logic using subgraph.
+    // Note: it would be nice if the client could be instantiated with dao identifier
+    // thereby removing the need to pass in the identifier for all the data retrieval methods
+
+    if (!daoIdentifier) {
+      throw new Error("A DAO identifier is needed");
+    }
+
+    // Generate DAO creation within the past year
+    const fromDate = new Date(
+      new Date().setFullYear(new Date().getFullYear() - 1)
+    ).getTime();
+
+    const dummyDaoNames = [
+      "Patito Dao",
+      "One World Dao",
+      "Sparta Dao",
+      "Yggdrasil Unite",
+    ];
+
+    return Promise.resolve({
+      ...(isAddress(daoIdentifier)
+        ? {
+            address: daoIdentifier,
+            name:
+              dummyDaoNames[
+                Math.floor(Math.random() * dummyDaoNames.length - 1)
+              ],
+          }
+        : {
+            address: "0x663ac3c648548eb8ccd292b41a8ff829631c846d",
+            name: daoIdentifier,
+          }),
+
+      createdAt: new Date(
+        fromDate + Math.random() * (Date.now() - fromDate)
+      ).getTime(),
+      description: `We are a community that loves trees and the planet. We track where forestation
+       is increasing (or shrinking), fund people who are growing and protecting trees...`,
+      links: [
+        {
+          label: "Website",
+          url: "https://google.com",
+        },
+        {
+          label: "Discord",
+          url: "https://google.com",
+        },
+      ],
+      packages: ["WhitelistPackage", "ERC20VotingPackage"],
+      token: {
+        address: "0x0000000000000000000000000000000000000000",
+        name: "Ethereum (Canonical)",
+        symbol: "ETH",
+        decimals: 18,
+      },
     });
   }
 }
 
 //// PRIVATE HELPERS
 
 function unwrapCreateDaoParams(
-  params: ICreateParams,
-): [
-  DAOFactory.DAOConfigStruct,
-  DAOFactory.VoteConfigStruct,
-  string,
-  string,
-] {
+  params: ICreateParams
+): [DAOFactory.DAOConfigStruct, DAOFactory.VoteConfigStruct, string, string] {
   // TODO: Serialize plugin params into a buffer
-  const pluginDataBytes = "0x" + params.plugins.map((entry) => {
-    const item = solidityPack(["uint256", "bytes[]"], [entry.id, entry.data]);
-    return strip0x(item);
-  }).join("");
+  const pluginDataBytes =
+    "0x" +
+    params.plugins
+      .map(entry => {
+        const item = solidityPack(
+          ["uint256", "bytes[]"],
+          [entry.id, entry.data]
+        );
+        return strip0x(item);
+      })
+      .join("");
```

Comment:
Having conflicting code formatters is an issue. I suggest using `deno fmt` (plugin available). Which is yours?